### ISSUE TITLE
[Bug]: search failed with error `segment lacks[segment=xx]: channel not available[channel=by-dev-rootcoord-xx]` after querynode pod kill chaos test

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#31481]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.4-20240320-a07abba1-amd64
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2024-03-21T00:13:45.320Z] <name>: Checker__lJAaPo78

[2024-03-21T00:13:45.320Z] <description>: 

[2024-03-21T00:13:45.320Z] <schema>: {'auto_id': False, 'description': '', 'fields': [{'name': 'int64', 'description': '', 'type': <DataType.INT64: 5>, 'is_primary': True, 'auto_id': False}, {'name': 'float', 'description': '', 'type': <DataType.FLOAT: 10>}......  (api_request.py:37)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:09:52 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:07 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:07 - DEBUG - ci_test]: (api_request)  : [Collection.compact] args: [180], kwargs: {} (api_request.py:62)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:07 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:08 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:09 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:09 - INFO - ci_test]: assert create collection: 0.00973057746887207, init_entities: 1155000 (test_all_collections_after_chaos.py:49)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:09 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [[[-3000, -2999, -2998, -2997, -2996, -2995, -2994, -2993, -2992, -2991, -2990, -2989, -2988, -2987, -2986, -2985, -2984, -2983, -2982, -2981, -2980, -2979, -2978, -2977, -2976, -2975, -2974, -2973, -2972, -2971, -2970, -2969, -2968, -2967, -2966, -2965, -2964, -2963, -2962, -2961, -2960, -2959, -29......, kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:09 - DEBUG - ci_test]: (api_response) : (insert count: 2000, delete count: 0, upsert count: 0, timestamp: 448523091112099842, success count: 2000, err count: 0)  (api_request.py:37)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:09 - INFO - ci_test]: assert insert: 0.2589986324310303 (test_all_collections_after_chaos.py:61)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:09 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:13 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:13 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:14 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:14 - INFO - ci_test]: assert flush: 4.598307132720947, entities: 1157000 (test_all_collections_after_chaos.py:71)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:14 - INFO - ci_test]: index info: [{'collection': 'Checker__lJAaPo78', 'field': 'float_vector', 'index_name': 'vec_index', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}] (test_all_collections_after_chaos.py:87)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:14 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:14 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:14 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.14540577250244002, 0.14178128989685151, 0.14205850625922484, 0.07224393785327152, 0.026947609578696404, 0.0888103812912946, 0.09185448581357535, 0.1433714284748155, 0.12740279623885614, 0.08599733890773484, 0.1323969829341291, 0.03180982519604184, 0.09757349708421843, 0.03263497844066052, 0.135......, kwargs: {} (api_request.py:62)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:14 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=503, message=failed to search: segment lacks[segment=448522566555280054]; segment lacks[segment=448522566555279388]: channel not available[channel=by-dev-rootcoord-dml_8_448522566555479408v1])>, <Time:{'RPC start': '2024-03-21 00:10:14.577621', 'RPC error': '2024-03-21 00:10:14.582378'}> (decorators.py:139)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:14 - ERROR - ci_test]: Traceback (most recent call last):

[2024-03-21T00:13:45.321Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 32, in inner_wrapper

[2024-03-21T00:13:45.321Z]     res = func(*args, **_kwargs)

[2024-03-21T00:13:45.321Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 63, in api_request

[2024-03-21T00:13:45.321Z]     return func(*arg, **kwargs)

[2024-03-21T00:13:45.321Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 792, in search

[2024-03-21T00:13:45.321Z]     resp = conn.search(

[2024-03-21T00:13:45.321Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 140, in handler

[2024-03-21T00:13:45.321Z]     raise e from e

[2024-03-21T00:13:45.321Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 136, in handler

[2024-03-21T00:13:45.321Z]     return func(*args, **kwargs)

[2024-03-21T00:13:45.321Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 175, in handler

[2024-03-21T00:13:45.321Z]     return func(self, *args, **kwargs)

[2024-03-21T00:13:45.321Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 115, in handler

[2024-03-21T00:13:45.321Z]     raise e from e

[2024-03-21T00:13:45.321Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 86, in handler

[2024-03-21T00:13:45.321Z]     return func(*args, **kwargs)

[2024-03-21T00:13:45.321Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 775, in search

[2024-03-21T00:13:45.321Z]     return self._execute_search(request, timeout, round_decimal=round_decimal, **kwargs)

[2024-03-21T00:13:45.321Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 735, in _execute_search

[2024-03-21T00:13:45.321Z]     raise e from e

[2024-03-21T00:13:45.321Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 728, in _execute_search

[2024-03-21T00:13:45.321Z]     check_status(response.status)

[2024-03-21T00:13:45.321Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/utils.py", line 54, in check_status

[2024-03-21T00:13:45.321Z]     raise MilvusException(status.code, status.reason, status.error_code)

[2024-03-21T00:13:45.321Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=503, message=failed to search: segment lacks[segment=448522566555280054]; segment lacks[segment=448522566555279388]: channel not available[channel=by-dev-rootcoord-dml_8_448522566555479408v1])>

[2024-03-21T00:13:45.321Z]  (api_request.py:45)

[2024-03-21T00:13:45.321Z] [2024-03-21 00:10:14 - ERROR - ci_test]: (api_response) : <MilvusException: (code=503, message=failed to search: segment lacks[segment=448522566555280054]; segment lacks[segment=448522566555279388]: channel not available[channel=by-dev-rootcoord-dml_8_448522566555479408v1])> (api_request.py:46)
```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-kafka-for-release-cron/detail/chaos-test-kafka-for-release-cron/11735/pipeline

log: 
[artifacts-querynode-pod-kill-11735-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14693441/artifacts-querynode-pod-kill-11735-server-logs.tar.gz)

**Comment by weiliu1031** [comment]
please verify this with latest image

**Comment by zhuwenxing** [comment]
verified and fixed in `2.4-20240327-d37e1fdd-amd64`
